### ISSUE TITLE
[bug] Case label selector mismatch

### ISSUE THREAD

**Issue by likeamahoney** [black-parrot/black-parrot#1230]
I'm doing some research related to static analysis of verilog and I found an interesting case that looks like a copy-paste bug.

[Here](https://github.com/black-parrot/black-parrot/blob/08edfb479c7b5d55d665a63464785044c6b4cd35/bp_fe/src/v/bp_fe_icache.sv#L582) label with is equal to 4 bit which do not match with selector case (`opcode`) width which is 3. 

Is this done on purpose or is it a mistake or typo or copy-paste bug from [here](https://github.com/black-parrot/black-parrot/blob/08edfb479c7b5d55d665a63464785044c6b4cd35/bp_be/src/v/bp_be_dcache/bp_be_dcache.sv#L882)?

**Comment by dpetrisko** [comment]
Yeah I’m sure that’s what happened. There is no tag_mem_fast_write condition in icache, so the case statement wasn’t modified to select based on it

Note that the selection is zero-extended by the standard so this is safe. Feel free to make a pr to clean it up, though
